tests/test_products/test_category_routers.py

```diff
@@ -35,52 +35,62 @@ def test_authenticated_user_get_single_category(
     response = sync_client.get(
         f"categories/{db_categories[1].id}", headers=auth_headers
     )
-    assert response.json()["id"] == db_categories[1].id
     assert response.status_code == status.HTTP_200_OK
+    assert response.json()["id"] == db_categories[1].id
 
 
-def test_authenticated_user_can_update_category(
+def test_superuser_can_update_category(
     sync_client: TestClient,
-    auth_headers: dict[str, str],
+    superuser_auth_headers: dict[str, str],
     db_categories: list[CategoryOutputSchema],
 ):
     update_data = CategoryInputSchemaFactory.build()
     response = sync_client.patch(
         f"categories/{db_categories[0].id}",
         data=update_data.json(),
-        headers=auth_headers,
+        headers=superuser_auth_headers,
     )
+
+    assert response.status_code == status.HTTP_200_OK
     assert response.json()["name"] == update_data.name
 
 
-def test_authenticated_user_can_delete_category(
+def test_superuser_can_delete_category(
     sync_client: TestClient,
-    auth_headers: dict[str, str],
+    superuser_auth_headers: dict[str, str],
     db_categories: list[CategoryOutputSchema],
 ):
     response = sync_client.delete(
-        f"categories/{db_categories[0].id}", headers=auth_headers
+        f"categories/{db_categories[0].id}", headers=superuser_auth_headers
     )
     assert response.status_code == status.HTTP_204_NO_CONTENT
 
 
-def test_authenticated_user_can_delete_all_categories(
+def test_superuser_can_delete_all_categories(
     sync_client: TestClient,
-    auth_headers: dict[str, str],
+    superuser_auth_headers: dict[str, str],
     db_categories: list[CategoryOutputSchema],
 ):
-    response = sync_client.delete("categories/", headers=auth_headers)
+    response = sync_client.delete("categories/", headers=superuser_auth_headers)
     assert response.status_code == status.HTTP_204_NO_CONTENT
 
-    response = sync_client.get("categories/", headers=auth_headers)
+    response = sync_client.get("categories/", headers=superuser_auth_headers)
+    assert response.status_code == status.HTTP_200_OK
     assert response.json()["total"] == 0
 
 
+def test_authenticated_user_cannot_get_categories(
+    sync_client: TestClient, auth_headers: dict[str, str]
+):
+    response = sync_client.get("categories/", headers=auth_headers)
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
```

Comment:
need to rethink if there's any point to enable to user getting categories , as category object only contains category name so this is not very userful to user